xarray/tests/test_coding.py

```diff
@@ -65,13 +65,13 @@ def test_CFMaskCoder_missing_value() -> None:
     expected.attrs["missing_value"] = -9999
 
     decoded = xr.decode_cf(expected.to_dataset())
-    encoded, _ = xr.conventions.cf_encoder(decoded, decoded.attrs)
+    encoded, _ = xr.conventions.cf_encoder(decoded.variables, decoded.attrs)
```

Comment:
@Illviljan This looks like a typing opportunity for a future PR.

Reply:
Sooo many to choose from! :D